src/sparseml/pytorch/image_classification/utils/trainer.py

```diff
@@ -233,8 +233,14 @@ def _apply_one_shot(self):
         _LOGGER.info(f"Applied {self.recipe_path} to manager")
 
     def _initialize_module_tester(self):
+        """
+        This uses self.model.module unlike the ModuleTrainer which uses self.model
+        so that the validation runs on the local copy of the model on the rank 0 gpu
+        rather than on the distributed model
+        """
+        
         tester = ModuleTester(
-            module=self.model,
+            module=self.model.module,
```

Comment:
this would fail if the model is not DP/DDP right?
let's add a check for `is_parallel_module` (https://github.com/neuralmagic/sparseml/blob/bbc61bec8fd250b0e94a89cbbb129b52da0eb542/src/sparseml/pytorch/utils/model.py#L355) before unwrapping

Reply:
fair point